astronomer/providers/snowflake/triggers/snowflake_trigger.py

```diff
@@ -141,9 +153,136 @@ async def is_still_running(self, query_id: str) -> bool:
         return False
         """
         hook = SnowflakeSqlApiHookAsync(
-            self.snowflake_conn_id, self.token_life_time, self.token_renewal_delta
+            self.snowflake_conn_id,
+            self.token_life_time,
+            self.token_renewal_delta,
         )
         statement_status = await hook.get_sql_api_query_status(query_id)
         if statement_status["status"] in ["running"]:
             return True
         return False
+
+
+class SnowflakeSensorTrigger(BaseTrigger):
+    """
+    This trigger validates the result of a query (asynchronously).
+    An Airflow Trigger asynchronously polls for a certain condition to be true (which yields a
+    ``TriggerEvent``), after which a synchronous piece of code can be used to complete the logic (set by
+    ``method_name`` on AsyncOperator/Sensor.defer()).
+    Docs: https://airflow.apache.org/docs/apache-airflow/stable/concepts/deferring.html#triggering-deferral
+    """
+
+    def __init__(
+        self,
+        sql: str,
+        dag_id: str,
+        task_id: str,
+        run_id: str,
+        snowflake_conn_id: str,
+        parameters: str = None,
+        success: str = None,
+        failure: str = None,
+        fail_on_empty: bool = False,
+        poke_interval: int = 60,
```

Comment:
Use float instead of int. Since ``BaseSensorOperator`` has attribute poke_interval: float = 60,.or else mypy will throw an error